tests/test_scripts.py

```diff
@@ -1926,3 +1926,15 @@ def test_readstats_empty():
     assert status == 0
 
     assert expected_output in out
+
+
+def test_existance_failure():
+    expected_output = 'ERROR: Input file'
+
+    args = ['thisfiledoesnotexistatall']
+
+    status, out, err = utils.runscript(
```

Comment:
for running a spell checker - what editor are you using? it'll depend on that. or you can just copy/paste the code into Word...

yes, I misread the test in test_scripts - it's good now!

the test in test_script_arguments should be of the specific kfile function, as opposed to running a script that runs the kfile function.  so, it should call kfile.check_file_status specifically, and verify that it raises a SystemExit exception when the file doesn't exist; and maybe another test to check that if you specify force, it doesn't exit; etc.  Basically you should test all of the if/else paths in the code.  Does that make more sense?